shared/utility/vision/Vision.cpp

```diff
@@ -186,42 +186,42 @@ namespace utility::vision {
         Eigen::Matrix<uint8_t, 5, 5> patch = getSubImage(x, y, width, height, data);
 
         // Work out what pixel type we are
-        const int row       = y % 2;
-        const int col       = x % 2;
-        BayerPixelType type = row   ? col ? BayerPixelType::R : BayerPixelType::GR
-                              : col ? BayerPixelType::GB
-                                    : BayerPixelType::B;
+        const unsigned int row = y % 2;
+        const unsigned int col = x % 2;
+        BayerPixelType type    = row   ? col ? BayerPixelType::R : BayerPixelType::GR
+                                 : col ? BayerPixelType::GB
+                                       : BayerPixelType::B;
 
         return getBayerPixel(patch, type);
     }
 
     Pixel getGrey16Pixel(uint x, uint y, int width, int /*height*/, const std::vector<uint8_t>& data) {
-        int origin = (y * width + x) * 2;
+        const unsigned int origin = (y * width + x) * 2;
```

Comment:
All these shoudl 100% be kept as regular ints not unsigned ints. Doing math for pixel coordinates in unsigned math is asking for trouble. Every so often an expression will dip below 0 and back up, but with unsigned it explodes and has memory issues.

In general, try to always use signed integers unless there is an explicit reason to use unsigned integers. For the most common case where it gets mad at you for looping on a .size() of something, use `int(x.size())` instead. Until we get c++20 when they add `x.ssize()`

Even the original authors of c++ consider it one of their blunders that .size() returns an unsigned integer